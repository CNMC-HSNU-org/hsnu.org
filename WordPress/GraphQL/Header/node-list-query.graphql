query NODE_LIST_QUERY($first: Int!, $after: String) {
  headers(first: $first, after: $after) {
    nodes {
      acf_header {
        fieldGroupName
        name
        url
      }
      content
      contentType {
        node {
          id
        }
      }
      databaseId
      date
      dateGmt
      desiredSlug
      enclosure
      guid
      id
      lastEditedBy {
        node {
          id
        }
      }
      link
      modified
      modifiedGmt
      slug
      status
      template {
        templateFile
        templateName
        __typename
      }
      title
      uri
      __typename
    }
    pageInfo {
      hasNextPage
      endCursor
    }
  }
}
