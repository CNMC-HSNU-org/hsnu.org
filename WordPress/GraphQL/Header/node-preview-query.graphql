query PREVIEW_QUERY($id: ID!) {
  header(id: $id, idType: ID, asPreview: true) {
    acf_header {
      fieldGroupName
      name
      url
    }
    content
    contentType {
      node {
        id
      }
    }
    databaseId
    date
    dateGmt
    desiredSlug
    enclosure
    guid
    id
    lastEditedBy {
      node {
        id
      }
    }
    link
    modified
    modifiedGmt
    slug
    status
    template {
      templateFile
      templateName
      __typename
    }
    title
    uri
    __typename
  }
}
