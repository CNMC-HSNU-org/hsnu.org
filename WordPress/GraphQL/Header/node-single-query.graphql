query SINGLE_CONTENT_QUERY($id: ID!) {
  header(id: $id) {
    acf_header {
      fieldGroupName
      name
      url
    }
    content
    contentType {
      node {
        id
      }
    }
    databaseId
    date
    dateGmt
    desiredSlug
    enclosure
    guid
    id
    lastEditedBy {
      node {
        id
      }
    }
    link
    modified
    modifiedGmt
    slug
    status
    template {
      templateFile
      templateName
      __typename
    }
    title
    uri
    __typename
  }
}
