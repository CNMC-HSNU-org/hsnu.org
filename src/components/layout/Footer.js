import React from "react"
import { Container, Row, Col } from "react-bootstrap"

// icons
import medium from "../../images/icons/socialMedia/medium.svg"
import twitter from "../../images/icons/socialMedia/twitter.svg"
import github from "../../images/icons/socialMedia/github.svg"
import patreon from "../../images/icons/socialMedia/patreon.svg"

const Footer = () => {
  const footers = [
    { name: "twitter", link: "https://twitter.com", icon: twitter },
    { name: "medium", link: "https://medium.com", icon: medium },
    { name: "github", link: "https://github.com/HSNU-ORG", icon: github },
    { name: "patreon", link: "https://patreon.com", icon: patreon },
  ]

  return (
    <>
      <footer>
        <Container fluid>
          <Row>
            <Col lg="6" className={"logo"}>
              <h2 className={"is-2 bold"}>
                Made by team <a href="https://hsnu.org">hsnu.org</a>
              </h2>
            </Col>
            <Col lg="6" className={"icons"}>
              {footers.map(icon => (
                <a
                  href={icon.link}
                  target="_blank"
                  rel="noopener noreferrer"
                  key={icon.name}
                >
                  <img src={icon.icon} alt={icon.name}></img>
                </a>
              ))}
            </Col>
          </Row>
        </Container>
      </footer>
    </>
  )
}

export default Footer
