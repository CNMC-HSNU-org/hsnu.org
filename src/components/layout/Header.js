import React from "react"
import { Navbar, Nav, Form, Button, FormControl } from "react-bootstrap"

// icon and images
import hsnu from "../../images/icons/HSNU.svg"
import menu from "../../images/icons/menu.svg"
import search from "../../images/icons/search.svg"

const Header = () => {
  return (
    <Navbar expand="xl" id="header">
      {/* Brand */}
      <Navbar.Brand href="#">
        <img src={hsnu} alt="Logo of HSNU"></img>
      </Navbar.Brand>

      {/* Toggler */}
      <Navbar.Toggle
        onClick={() => {
          const sidebar = document.querySelector("#sidebar")
          const overlay = document.querySelector("#sidebar-overlay")
          sidebar.classList.toggle("active")
          overlay.classList.toggle("active")
        }}
      />

      {/* Navbar Items */}
      <Navbar.Collapse>
        <Nav>
          {[1, 2, 3, 4].map(item => (
            <Nav.Item>
              <Nav.Link className={"is-7"}>Student</Nav.Link>
            </Nav.Item>
          ))}
          {/* search */}
          <Nav.Item
            onClick={() => {
              const navsearch = document.querySelector("#nav-search")
              const searbox = document.querySelector("#nav-search-box")
              navsearch.classList.toggle("active")
              if (navsearch.classList.contains("active")) {
                searbox.focus()
              }
            }}
          >
            <Nav.Link>
              <img src={search} />
            </Nav.Link>
          </Nav.Item>

          {/* menu */}
          <Nav.Item
            onClick={() => {
              const sidebar = document.querySelector("#sidebar")
              const overlay = document.querySelector("#sidebar-overlay")
              sidebar.classList.toggle("active")
              overlay.classList.toggle("active")
            }}
          >
            <Nav.Link>
              <img src={menu} />
            </Nav.Link>
          </Nav.Item>
        </Nav>
      </Navbar.Collapse>

      {/* search box in the nav */}
      <div id="nav-search">
        <Form>
          <FormControl
            id="nav-search-box"
            type="text"
            placeholder="Search"
            className={"is-4-regular"}
          />
        </Form>
      </div>
    </Navbar>
  )
}

export default Header
