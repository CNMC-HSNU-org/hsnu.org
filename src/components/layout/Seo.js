import React from "react"
import { Helmet } from "react-helmet"
import PropTypes from "prop-types"
import { useStaticQuery, graphql } from "gatsby"

const SEO = ({ title, description, pathname, article }) => {
  const meta = useStaticQuery(graphql`
    {
      site {
        siteMetadata {
          defaultTitle: title
          titleTemplate
          defaultDescription: description
          siteUrl: siteUrl
          image
        }
      }
    }
  `).site.siteMetadata

  const seo = {
    title: title || meta.defaultTitle,
    titleTemplate: meta.titleTemplate,
    description: description || meta.defaultDescription,
    url: `${meta.siteUrl}${pathname || "/"}`,
    image: meta.siteImage,
    article: article,
  }

  return (
    <>
      <Helmet title={seo.title} titleTemplate={seo.titleTemplate}>
        <meta charset="UTF-8" />
        <link rel="icon" href="src/images/icons/HSNU.png"></link>
        <meta
          name="keywords"
          content="HSNU, 師大附中, 附中, 臺北高中, 大安區"
        />
        <meta name="author" content="github@Anmicius0" />
        <meta name="description" content={seo.description} />
        <meta name="image" content={seo.image} />

        {/* open graph (FaceBook)*/}
        {seo.url && <meta property="og:url" content={seo.url} />}
        {(seo.article ? true : null) && (
          <meta property="og:type" content="article" />
        )}
        {seo.title && <meta property="og:title" content={seo.title} />}
        {seo.description && (
          <meta property="og:description" content={seo.description} />
        )}
        {seo.image && <meta property="og:image" content={seo.image} />}

        {/* twitter */}
        <meta name="twitter:card" content="summary_large_image" />
        {seo.title && <meta name="twitter:title" content={seo.title} />}
        {seo.description && (
          <meta name="twitter:description" content={seo.description} />
        )}
        {seo.image && <meta name="twitter:image" content={seo.image} />}

        {/* Google Search Console */}
        <meta
          name="google-site-verification"
          content="Wy3ED0joPSTy8mtnur6mrxByXxOhP6LT4sMrnu5p818"
        />
      </Helmet>
    </>
  )
}

export default SEO

SEO.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  pathname: PropTypes.string,
  article: PropTypes.bool,
}

SEO.defaultProps = {
  title: null,
  description: null,
  image: null,
  pathname: null,
  article: false,
}
