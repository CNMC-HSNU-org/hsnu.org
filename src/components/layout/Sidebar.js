import React from "react"
import { useStaticQuery, graphql } from "gatsby"

// tools
import SearchBox from "../tools/SearchBox/SearchBox"
import Endorsement from "../tools/Endorsement/Endorsement"

const Sidebar = () => {
  const menus = useStaticQuery(
    graphql`
      query {
        allMenuJson {
          edges {
            node {
              title
              items {
                title
                url
              }
            }
          }
        }
      }
    `
  ).allMenuJson.edges

  console.log(menus)

  return (
    <>
      <div id={"sidebar"}>
        {/* title */}
        <h3 className={"is-3 bold"}>國立臺灣大學附屬高級中學</h3>

        <Endorsement />
        <SearchBox />

        {/* menu cards*/}
        <ul className={"menu-cards"}>
          {menus.map(menu => (
            <li key={menu.node.title}>
              <ul className={"menu-card"}>
                <h4 className={"is-4 bold"}>{menu.node.title}</h4>
                {menu.node.items.map(item => (
                  <li className={"is-5"} key={item.title}>
                    <a href={item.url}>{item.title}</a>
                  </li>
                ))}
              </ul>
            </li>
          ))}
        </ul>
        {/* end of menucards */}
      </div>

      {/* overlay */}
      <div
        id={"sidebar-overlay"}
        onClick={() => {
          const sidebar = document.querySelector("#sidebar")
          const overlay = document.querySelector("#sidebar-overlay")
          const collapse = document.querySelector(".navbar-collapse")
          sidebar.classList.toggle("active")
          overlay.classList.toggle("active")
          collapse.classList.remove("show")
        }}
      ></div>
    </>
  )
}

export default Sidebar
