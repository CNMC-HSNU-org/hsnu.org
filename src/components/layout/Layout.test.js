import React from "react"
import renderer from "react-test-renderer"
import { HeaderPure } from "./Header"
import Footer from "./Footer"
import { SidebarPure } from "./Sidebar"
import { SEOPure } from "./Seo"

describe("Layout/Header", () => {
  it("renders correctly", () => {
    // mock navs
    const navs = [
      {
        node: {
          acf: {
            name: "iLearn",
            url: "https://ilearning.hs.ntnu.edu.tw/",
          },
        },
      },
      {
        node: {
          acf: {
            name: "學習歷程",
            url: "https://e-portfolio.cooc.tp.edu.tw/Portal.do",
          },
        },
      },
      {
        node: {
          acf: {
            name: "報修",
            url: "http://artlife.hs.ntnu.edu.tw/case122/index.php",
          },
        },
      },
      {
        node: {
          acf: {
            name: "二代校務",
            url: "https://sschool.tp.edu.tw/Login.action",
          },
        },
      },
    ]

    const tree = renderer.create(<HeaderPure navs={navs} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("Layout/Footer", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Footer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("Layout/Sidebar", () => {
  it("renders correctly", () => {
    const menus = [
      {
        node: {
          title: "招考專區",
          items: [
            {
              title: "教師甄選作業系統",
              url: "http://artlife.hs.ntnu.edu.tw/case43a/",
            },
            {
              title: "109年科學班招生專區",
              url:
                "https://site.hs.ntnu.edu.tw/science-class/%e6%8b%9b%e7%94%9f%e5%b0%88%e5%8d%80-2/",
            },
            {
              title: "109年資訊班招生專區",
              url: "https://site.hs.ntnu.edu.tw/cs-class/",
            },
            {
              title: "109學測及二次英聽測驗校內網路報名系統",
              url: "http://140.131.149.106/",
            },
            {
              title: "108年國中音樂班聯招專區",
              url: "https://site.hs.ntnu.edu.tw/music108s/",
            },
          ],
        },
      },
      {
        node: {
          title: "校內資訊",
          items: [
            {
              title: "行事曆",
              url: "http://artlife.hs.ntnu.edu.tw/case43a/",
            },
            {
              title: "網路使用",
              url:
                "https://site.hs.ntnu.edu.tw/science-class/%e6%8b%9b%e7%94%9f%e5%b0%88%e5%8d%80-2/",
            },
            {
              title: "校慶",
              url: "https://site.hs.ntnu.edu.tw/cs-class/",
            },
            {
              title: "課表",
              url: "http://140.131.149.106/",
            },
            {
              title: "菜單",
              url: "https://site.hs.ntnu.edu.tw/music108s/",
            },
          ],
        },
      },
      {
        node: {
          title: "學生資訊",
          items: [
            {
              title: "升學資訊",
              url:
                "https://site.hs.ntnu.edu.tw/guidance/%E6%9C%80%E6%96%B0%E6%B6%88%E6%81%AF2019/",
            },
            {
              title: "微課程",
              url:
                "https://www.hs.ntnu.edu.tw/%e5%be%ae%e8%aa%b2%e7%a8%8b%e8%b3%87%e8%a8%8a/",
            },
            {
              title: "獎學金資訊",
              url:
                "https://www.hs.ntnu.edu.tw/search/?type=bt&sq=%E7%8D%8E%E5%AD%B8%E9%87%91",
            },
            {
              title: "學生證掛失",
              url:
                "https://ecardlost.tp.edu.tw/ecard/Ecard/QueryOfLostEcardAction",
            },
            {
              title: "證明文件申請",
              url:
                "https://site.hs.ntnu.edu.tw/academic/hsnureg/%E7%94%B3%E8%AB%8B%E6%B5%81%E7%A8%8B%E5%8F%8A%E8%A1%A8%E5%96%AE/",
            },
          ],
        },
      },
      {
        node: {
          title: "相關單位",
          items: [
            {
              title: "校友會",
              url: "https://hsnuaa.org.tw/",
            },
            {
              title: "家長會",
              url:
                "https://sites.google.com/a/pccu.edu.tw/guo-li-shi-da-fu-zhong72jie-jia-zhang-hui/guan-yu-ben-hui?previewAsViewer=1",
            },
            {
              title: "志工團",
              url: "https://site.hs.ntnu.edu.tw/ilove143/",
            },
            {
              title: "教師會",
              url: "https://site.hs.ntnu.edu.tw/teachers/",
            },
            {
              title: "合作社",
              url: "https://site.hs.ntnu.edu.tw/cooperative/",
            },
          ],
        },
      },
      {
        node: {
          title: "宣傳內容",
          items: [
            {
              title: "捲款專區",
              url: "https://www.hs.ntnu.edu.tw/support/",
            },
            {
              title: "政策宣導",
              url: "https://www.hs.ntnu.edu.tw/links/advocacy-links/",
            },
            {
              title: "108課綱",
              url:
                "https://www.hs.ntnu.edu.tw/108%E8%AA%B2%E7%B6%B1%E7%9B%B8%E9%97%9C/",
            },
            {
              title: "教師會",
              url: "https://site.hs.ntnu.edu.tw/teachers/",
            },
            {
              title: "合作社",
              url: "https://site.hs.ntnu.edu.tw/cooperative/",
            },
          ],
        },
      },
    ]

    const tree = renderer.create(<SidebarPure menus={menus} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
