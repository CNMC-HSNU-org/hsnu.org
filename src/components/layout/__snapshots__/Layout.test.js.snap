// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/Footer renders correctly 1`] = `
<footer>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="logo col-lg-6"
      >
        <h2
          className="is-2 bold"
        >
          We Are HSNUer
        </h2>
      </div>
      <div
        className="icons col-lg-6"
      >
        <a
          href="https://twitter.com"
        >
          <img
            alt="twitter"
            src="test-file-stub"
          />
        </a>
        <a
          href="https://medium.com"
        >
          <img
            alt="medium"
            src="test-file-stub"
          />
        </a>
        <a
          href="https://github.com"
        >
          <img
            alt="github"
            src="test-file-stub"
          />
        </a>
        <a
          href="https://patreon.com"
        >
          <img
            alt="patreon"
            src="test-file-stub"
          />
        </a>
      </div>
    </div>
  </div>
</footer>
`;

exports[`Layout/Header renders correctly 1`] = `
<nav
  className="navbar navbar-expand-xl navbar-light"
  id="header"
>
  <a
    className="navbar-brand"
    href="/"
  >
    <img
      alt="師大附中"
      src="test-file-stub"
    />
  </a>
  <button
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
      >
        <a
          className="is-5 bold nav-link"
          data-rb-event-key="https://ilearning.hs.ntnu.edu.tw/"
          href="https://ilearning.hs.ntnu.edu.tw/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          iLearn
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="is-5 bold nav-link"
          data-rb-event-key="https://e-portfolio.cooc.tp.edu.tw/Portal.do"
          href="https://e-portfolio.cooc.tp.edu.tw/Portal.do"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          學習歷程
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="is-5 bold nav-link"
          data-rb-event-key="http://artlife.hs.ntnu.edu.tw/case122/index.php"
          href="http://artlife.hs.ntnu.edu.tw/case122/index.php"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          報修
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="is-5 bold nav-link"
          data-rb-event-key="https://sschool.tp.edu.tw/Login.action"
          href="https://sschool.tp.edu.tw/Login.action"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          二代校務
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          className="is-5 bold nav-link"
          data-rb-event-key="/english"
          href="/english"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          English
        </a>
      </div>
      <div
        className="nav-item"
        onClick={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <img
            alt="search"
            src="test-file-stub"
          />
        </a>
      </div>
      <div
        className="nav-item"
        onClick={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <img
            alt="menu"
            src="test-file-stub"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    id="nav-search"
  >
    <form
      action="/search"
      className=""
      method="get"
    >
      <input
        className="is-3 bold form-control"
        id="nav-search-box"
        name="search"
        placeholder="Search"
        type="text"
      />
    </form>
  </div>
</nav>
`;

exports[`Layout/SEO renders correctly 1`] = `null`;

exports[`Layout/Sidebar renders correctly 1`] = `
Array [
  <div
    id="sidebar"
  >
    <h3
      className="is-3 bold"
    >
      國立臺灣大學附屬高級中學
    </h3>
    <div
      className="endorsement"
    >
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "#FF4040",
          }
        }
      />
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "#FFAF37",
          }
        }
      />
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "#FBFF32",
          }
        }
      />
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "#A0FFA4",
          }
        }
      />
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "#76DEFF",
          }
        }
      />
      <div
        className="dot"
        style={
          Object {
            "backgroundColor": "#874EFF",
          }
        }
      />
    </div>
    <form
      action="/search"
      className="search-box"
      method="get"
    >
      <label
        className="form-label"
        hidden={true}
      >
        Search Box
      </label>
      <input
        className="is-4 form-control"
        name="search"
        placeholder="Search"
        type="text"
      />
      <button
        className="btn btn-light"
        disabled={false}
        type="submit"
      >
        <img
          alt="search!"
          src="test-file-stub"
        />
      </button>
    </form>
    <ul
      className="menu-cards"
    >
      <li>
        <ul
          className="menu-card"
        >
          <h4
            className="is-4 bold"
          >
            招考專區
          </h4>
          <li
            className="is-5"
          >
            <a
              href="http://artlife.hs.ntnu.edu.tw/case43a/"
            >
              教師甄選作業系統
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/science-class/%e6%8b%9b%e7%94%9f%e5%b0%88%e5%8d%80-2/"
            >
              109年科學班招生專區
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/cs-class/"
            >
              109年資訊班招生專區
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="http://140.131.149.106/"
            >
              109學測及二次英聽測驗校內網路報名系統
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/music108s/"
            >
              108年國中音樂班聯招專區
            </a>
          </li>
        </ul>
      </li>
      <li>
        <ul
          className="menu-card"
        >
          <h4
            className="is-4 bold"
          >
            校內資訊
          </h4>
          <li
            className="is-5"
          >
            <a
              href="http://artlife.hs.ntnu.edu.tw/case43a/"
            >
              行事曆
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/science-class/%e6%8b%9b%e7%94%9f%e5%b0%88%e5%8d%80-2/"
            >
              網路使用
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/cs-class/"
            >
              校慶
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="http://140.131.149.106/"
            >
              課表
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/music108s/"
            >
              菜單
            </a>
          </li>
        </ul>
      </li>
      <li>
        <ul
          className="menu-card"
        >
          <h4
            className="is-4 bold"
          >
            學生資訊
          </h4>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/guidance/%E6%9C%80%E6%96%B0%E6%B6%88%E6%81%AF2019/"
            >
              升學資訊
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://www.hs.ntnu.edu.tw/%e5%be%ae%e8%aa%b2%e7%a8%8b%e8%b3%87%e8%a8%8a/"
            >
              微課程
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://www.hs.ntnu.edu.tw/search/?type=bt&sq=%E7%8D%8E%E5%AD%B8%E9%87%91"
            >
              獎學金資訊
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://ecardlost.tp.edu.tw/ecard/Ecard/QueryOfLostEcardAction"
            >
              學生證掛失
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/academic/hsnureg/%E7%94%B3%E8%AB%8B%E6%B5%81%E7%A8%8B%E5%8F%8A%E8%A1%A8%E5%96%AE/"
            >
              證明文件申請
            </a>
          </li>
        </ul>
      </li>
      <li>
        <ul
          className="menu-card"
        >
          <h4
            className="is-4 bold"
          >
            相關單位
          </h4>
          <li
            className="is-5"
          >
            <a
              href="https://hsnuaa.org.tw/"
            >
              校友會
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://sites.google.com/a/pccu.edu.tw/guo-li-shi-da-fu-zhong72jie-jia-zhang-hui/guan-yu-ben-hui?previewAsViewer=1"
            >
              家長會
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/ilove143/"
            >
              志工團
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/teachers/"
            >
              教師會
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/cooperative/"
            >
              合作社
            </a>
          </li>
        </ul>
      </li>
      <li>
        <ul
          className="menu-card"
        >
          <h4
            className="is-4 bold"
          >
            宣傳內容
          </h4>
          <li
            className="is-5"
          >
            <a
              href="https://www.hs.ntnu.edu.tw/support/"
            >
              捲款專區
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://www.hs.ntnu.edu.tw/links/advocacy-links/"
            >
              政策宣導
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://www.hs.ntnu.edu.tw/108%E8%AA%B2%E7%B6%B1%E7%9B%B8%E9%97%9C/"
            >
              108課綱
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/teachers/"
            >
              教師會
            </a>
          </li>
          <li
            className="is-5"
          >
            <a
              href="https://site.hs.ntnu.edu.tw/cooperative/"
            >
              合作社
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>,
  <div
    id="sidebar-overlay"
    onClick={[Function]}
  />,
]
`;
